public/modules/custom/hel_tpm_general/src/Access/GroupNodeCreateAccessService.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\hel_tpm_general\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Path\PathValidatorInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Drupal\group\Access\GroupContentCreateEntityAccessCheck;
+use Drupal\group\Entity\GroupInterface;
+use Drupal\group\GroupMembershipLoader;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class GroupNodeCreateAccessService {
+
+  /**
+   * @var \Drupal\group\GroupMembershipLoader
+   */
+  protected $groupMembershipLoader;
+
+  /**
+   * @var \Drupal\group\Access\GroupContentCreateEntityAccessCheck
+   */
+  protected $groupContentCreateEntityAccess;
+
+  /**
+   * @var \Symfony\Component\Routing\RouterInterface
+   */
+  protected $router;
+
+  /**
+   * @param \Drupal\group\GroupMembershipLoader $group_membership_loader
+   * @param \Drupal\group\Access\GroupContentCreateEntityAccessCheck $group_content_create_entity_access
+   * @param \Drupal\Core\Path\PathValidatorInterface $path_validator
+   * @param \Symfony\Component\Routing\RouterInterface $router
+   */
+  public function __construct(GroupMembershipLoader $group_membership_loader, GroupContentCreateEntityAccessCheck $group_content_create_entity_access, RouterInterface $router) {
+    $this->groupMembershipLoader = $group_membership_loader;
+    $this->groupContentCreateEntityAccess = $group_content_create_entity_access;
+    $this->router = $router;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('group.membership_loader'),
+      $container->get('access_check.group_content.create_entity'),
+      $container->get('router.no_access_checks')
+    );
+  }
+
+  /**
+   * Access callback.
+   *
+   * @return \Drupal\Core\Access\AccessResultAllowed|\Drupal\Core\Access\AccessResultForbidden
+   */
+  public function hasAccess(AccountInterface $account, $plugin_id) {
+    $groups = $this->userGroups($account);
+    if (empty($groups)) {
```

Comment:
This is unnecessary; if $groups is empty, the foreach() will be skipped and forbidden returned anyway.